From 1656a1628c1d6ab81b0126c3c4540b2becd0558d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tim=20R=C3=BChsen?= <tim.ruehsen@gmx.de>
Date: Mon, 29 Jun 2020 18:04:48 +0200
Subject: [PATCH] * src/ftp.c (ftp_loop_internal): Check for VERIFCERTERR to
 avoid SIGABRT

There is a bug that causes wget to exit with SIGABRT when trying to
receive files through FTP from a server with a certificate that failed
the verification.

The bug is filed in RedHat Bugzilla for Fedora:
https://bugzilla.redhat.com/show_bug.cgi?id=1475861

Reported-by: Artem Egorenkov <aegorenk@redhat.com>
---
 src/ftp.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/ftp.c b/src/ftp.c
index 2b055376..37811451 100644
--- a/src/ftp.c
+++ b/src/ftp.c
@@ -1984,6 +1984,7 @@ ftp_loop_internal (struct url *u, struct url *original_url, struct fileinfo *f,
         case HOSTERR: case CONIMPOSSIBLE: case FWRITEERR: case FOPENERR:
         case FTPNSFOD: case FTPLOGINC: case FTPNOPASV: case FTPNOAUTH: case FTPNOPBSZ: case FTPNOPROT:
         case UNLINKERR: case WARC_TMP_FWRITEERR: case CONSSLERR: case CONTNOTSUPPORTED:
+        case VERIFCERTERR:
 #ifdef HAVE_SSL
           if (err == FTPNOAUTH)
             logputs (LOG_NOTQUIET, "Server does not support AUTH TLS.\n");
-- 
2.25.1

